job_runner.py

```diff
@@ -124,6 +124,58 @@ def welcome_proton(user):
     )
 
 
+def delete_mailbox_job(job: Job):
+    mailbox_id = job.payload.get("mailbox_id")
+    mailbox = Mailbox.get(mailbox_id)
+    if not mailbox:
+        return
+
+    transfer_mailbox_id = job.payload.get("transfer_mailbox_id")
+    alias_transferred_to = None
+    if transfer_mailbox_id:
+        transfer_mailbox = Mailbox.get(transfer_mailbox_id)
+        if transfer_mailbox:
+            alias_transferred_to = transfer_mailbox.email
+
+            for alias in mailbox.aliases:
+                if alias.mailbox_id == mailbox.id:
+                    alias.mailbox_id = transfer_mailbox.id
+                    if transfer_mailbox in alias._mailboxes:
+                        alias._mailboxes.remove(transfer_mailbox)
+                else:
+                    alias._mailboxes.remove(mailbox)
+                    if transfer_mailbox not in alias._mailboxes:
```

Comment:
can that happen that `alias.mailbox_id = transfer_mailbox_id`?

Reply:
I am not sure to understand your question:

If the mailbox we want to delete, is the "main" mailbox of the alias, we set the "main" mailbox to the **transfer_mailbox** and the value is saved with Session.commit()

Reply:
It's validated in the request before creating the job that they aren't the same.

Reply:
Ha ok, I understand the question now. Yeah, I agree with @acasajus.